contracts/cw3-flex-multisig/src/state.rs

```diff
@@ -45,6 +86,53 @@ impl Proposal {
 
         status
     }
+
+    /// update_status sets the status of the proposal to current_status.
+    /// (designed for handler logic)
+    pub fn update_status(&mut self, block: &BlockInfo) {
+        self.status = self.current_status(block);
+    }
+
+    // returns true iff this proposal is sure to pass (even before expiration if no future
+    // sequence of possible votes can cause it to fail)
+    pub fn is_passed(&self, block: &BlockInfo) -> bool {
+        match self.threshold {
+            Threshold::AbsoluteCount { weight_needed } => self.votes.yes >= weight_needed,
+            Threshold::AbsolutePercentage { percentage_needed } => {
+                self.votes.yes >= votes_needed(self.total_weight, percentage_needed)
+            }
+            Threshold::ThresholdQuora { threshold, quroum } => {
+                // this one is tricky, as we have two compares:
+                if self.expires.is_expired(block) {
+                    // * if we have closed yet, we need quorum% of total votes to have voted (counting abstain)
+                    //   and threshold% of yes votes from those who voted (ignoring abstain)
```

Comment:
Why ignoring abstain here, and not also above in `AbsolutePercentage`?

Reply:
AbsolutePercentage is 50% of the total weight of the group.

In quorum, we care about the ones who vote, as long as enough vote.

Reply:
Same rules apply, IMO. Suppose there are 7 voters, absolute percentage is 50%, 3 vote yes, and one abstains. Proposal should pass, right?

OK, I'll read the document, and continue from here.